lib/ansible/cli/doc.py

```diff
@@ -350,6 +350,14 @@ def _get_plugin_doc(plugin, plugin_type, loader, search_paths):
         if doc is None:
             raise ValueError('%s did not contain a DOCUMENTATION attribute' % plugin)
 
+        # generate extra data
+        if plugin_type == 'module':
+            # is there corresponding action plugin?
+            if plugin in action_loader:
+                doc['has_action'] = True
+            else:
+                doc['has_action'] = False
+
```

Comment:
It's been quite a few months since I looked at this but I think this belongs in format_plugin_docs.

The code should be organized roughly along the lines of the six stages of data processing: https://medium.com/@peerxp/the-6-stages-of-data-processing-cycle-3c2927c466ff  Since this is taking raw data (a plugin_name and the list of action names) and creating a derived value from it, it should be done in the processing stage rather than the input stage.

Reply:
Yeah, looking at the code, I remember this.... the output phase was really, really screwed up before (intertwined with the data processing phase) and we were on a deadline so I didn't fully decouple the output phase when I restructured the run() method.  There should be another explict phase (processing) here: https://github.com/ansible/ansible/pull/72359/files#diff-9732216f79978750789dbeb4c2ed1280d841c4211306ecb1b5c70e3db3fcb2f7R263

All of the generation of the stuff in the generate extra data section of `format_plugin_doc` should go into that phase.

Reply:
I think the other stuff in `format_plugin_doc` (`nowdate`, `docuri`) is actually not used at all and can be deleted.